### ISSUE TITLE
Question: order of components and outlines?

### ISSUE THREAD

**Issue by madig** [unified-font-object/ufo-spec#169]
Is there any implicit or explicit ordering requirement when dealing with outlines and components? 

1. If a glyph contains both outlines and components, does it matter what you draw into a TTF/CFF glyph first, the outlines or the (decomposed) components?
2. In case your components are mixed as well, does it matter if you draw nested outlines first or first descend into the nested components or any other strategy?
3. Does it matter if the decomposed components are added as outlines in whatever order is easiest to implement or should decomposition retain the implicit order, i.e. always go from first to last element in the component list?

To put it differently, does it matter if the sequence `[[["a", 1, 2], ["b", "c", 3, 4]], [[5, 6], ["d", 7, 8]]]` is randomly flattened into `["b", 4, 3, "d", 1, "a", 8, "c", 6, 5, 7, 2]`?

**Comment by justvanrossum** [comment]
Isn't that up to the target? Things like TTF/CFF generally don't care about the contour order, only about winding direction.

Randomly shuffled contours still represent the same shape, similar to how a closed contour still represents the same shape after rotating the point array to a different starting point.

I think this is a data transformation that is outside of the scope of what UFO needs to have an opinion about.

**Comment by benkiel** [comment]
Agree with Just, it does not seem UFO’s place to have an implicit or explicit opinion about this. @madig are you running into a use case or issue with UFO being silent on this?

**Comment by madig** [comment]
Thanks for the input. I'm not running into any issue, just implementing a glyph decomposition function and wondering about things!

**Comment by madig** [comment]
So specifically, I had things like https://unifiedfontobject.org/versions/ufo3/glyphs/glif/#publictruetypeinstructions in mind, because I think there's a defined order to "flatten" nested TrueType composites so that bytecode instructions can adress their intended contours no matter which engine is running them. Just in case you decide to flatten a composite without necessarily invalidating all of the bytecode. I suppose this is a niche concern.
